feat(core): media library full app dialog and selection validation

jmswrnr opened the pull request:
### Description

This PR replaces #9666 and continues on from that work. It will load a full-app experience of the Media Library in the studio asset selection in a larger dialog box.

> @skogsmaskin : It will also be possible to run real-time validation of the selection in the asset selection model against the edited document with the new Rule.media validator.

<!--
What changes are introduced?
Why are these changes introduced?
What issue(s) does this solve? (with link, if possible)
-->

### What to review

<!--
What steps should the reviewer take in order to review?
What parts/flows of the application/packages/tooling is affected?
-->

### Testing

<!--
Did you add sufficient testing for this change?
If not, please explain how you tested this change and why it was not
possible/practical for writing an automated test
-->

### Notes for release

This upgrades the Media Library asset selection dialog to utilize the full-app experience of Media Library, enabling browsing by collections and upload sessions. Also, enabling the management of assets from the dialog such as editing aspects and collections.

<!--
Engineers do not need to worry about the final copy,
but they must provide the docs team with enough context on:

* What changed
* How does one use it (code snippets, etc)
* Are there limitations we should be aware of
* [internal] Does this affect the docs team? If so, please ask a member of that team for a review

If this is PR is a partial implementation of a feature and is not enabled by default or if
this PR does not contain changes that needs mention in the release notes (tooling chores etc),
please call this out explicitly by writing "Part of feature X" or "Not required" in this section.
-->

pedrobonamin reviewed (commented):
Changes look good to me, added some minor suggestions but none is blocking. 
Recommend we check the types changes.

juice49 commented on packages/sanity/src/core/i18n/bundles/studio.ts:
```diff
@@ -187,6 +187,12 @@ export const studioLocaleStrings = defineLocalesResources('studio', {
   'asset-sources.media-library.image.title': 'Media Library',
 
+  /** Info messages for the Media Library Asset Source  */
+  'asset-sources.media-library.info.provisioning':
```
Is this string being used? Studio doesn't provision libraries anymore, and it doesn't look like this PR uses the string. But I could be wrong!

skogsmaskin replied:
It can be removed.

jmswrnr commented on packages/sanity/src/core/form/studio/assetSourceMediaLibrary/types.ts:
```diff
@@ -31,11 +31,6 @@ export interface SanityMediaLibraryConfig {
-     * The version of the plugin API to use (routes served by the Media Library app will be prefixed with this version)
-     * @defaultValue 'v1'
-     */
-    pluginApiVersion: string
```
Cool I've made it optional and marked as `@deprecated` now - the value isn't used anymore following these changes.

jmswrnr commented:
Thanks @juice49 @pedrobonamin - I've addressed the comments here and tagged @skogsmaskin to double check with context on `pluginApiVersion` and the locale string

pedrobonamin reviewed (approved):
Thank you @jmswrnr it looks good to me !
